src/js/components/AppConfigJSONEditorComponent.jsx

```diff
@@ -0,0 +1,63 @@
+import AceEditor from "react-ace";
+import "brace/mode/json";
+import "brace/theme/monokai";
+import React from "react/addons";
+import Util from "../helpers/Util";
+
+var AppConfigJSONEditorComponent = React.createClass({
+  displayName: "AppConfigJSONEditorComponent",
+
+  propTypes: {
+    app: React.PropTypes.object,
+    onChange: React.PropTypes.func,
+    onError: React.PropTypes.func
+  },
+
+  getDefaultProps: function () {
+    return {
+      app: {id: ""},
```

Comment:
How about moving this to a function outside of the component, that returns the "schema" for the json editor?
Something like this:
```

function getEmptyAppJSON() {
  return {
    id: ""
  };
}

var AppConfigJSONEditorComponent = React.createClass({

// ...
getDefaultProps: function () {
  return {
    app: getEmptyAppJSON(),
   // ... 
   }
},

// ...
getPrettyPrintedDefinition: function () {
  let app = this.props.app == null
   ? getEmptyAppJSON()
   : this.props.app;
  return JSON.stringify(app, null, 2);
},
```
?